edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -1122,8 +1122,10 @@ impl Authenticator for DockerModuleRuntime {
 
 fn init_client(docker_url: &Url) -> Result<DockerClient<UrlConnector>> {
     // build the hyper client
-    let client =
-        Client::builder().build(UrlConnector::new(docker_url).context(ErrorKind::Initialization)?);
+    let client = Client::builder()
+        // we don't need connection pool'ing for local docker socket.
+        .max_idle_per_host(1)
```

Comment:
Does `0` work?

Reply:
Honestly... don't know why I picket `1`:). Looking more at the hyper code, I realized that I need to set `keep_alive(false)` as well to completely disable connection pool.